xonsh/built_ins.py

```diff
@@ -637,15 +637,41 @@ def run_subproc(cmds, captured=True):
             hist.last_cmd_out = output
     if not prev_is_proxy and hist.last_cmd_rtn > 0 and ENV.get('RAISE_SUBPROC_ERROR'):
         raise CalledProcessError(hist.last_cmd_rtn, aliased_cmd, output=output)
-    if captured:
+    if captured=='stdout':
         return output
+    elif captured=='object':
+        return CompletedProcess(prev_proc, output)
 
 
-def subproc_captured(*cmds):
+class CompletedProcess:
```

Comment:
`subprocess.CompletedProcess` class is nothing more than a simple wrapper around `args`, `returncode`, `stdout` and `stderr` attributes. See the source code: https://github.com/python/cpython/blob/master/Lib/subprocess.py#L639

I don't see any problem in using it directly. But if you are afraid of that because it is undocumented and for example can change without warning, the whole class code can be copied as well.

To be precise, I don't like the present solution, because `stdout` and `stderr` attributes of your `CompletedProcess` has different meaning than `stdout` and `stderr` attributes of `subprocess.CompletedProcess`. In `subprocess.CompletedProcess` these are strings/bytes containing captured contents of respective streams. In your `CompletedProcess` class these are streams itself, which would need to be `.read()` to access their contents. But this would be pointless anyway, as `stdout` was already read to get `output`, so it is empty. And `stderr` isn't PIPEd on `Popen` creation, so it is empty as well.